File: app/src/main/java/com/sevtinge/cemiuiler/module/securitycenter/lab/GetNumberEnable.java
package com.sevtinge.cemiuiler.module.securitycenter.lab;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import com.sevtinge.cemiuiler.module.securitycenter.SecurityCenterDexKit;
import com.sevtinge.cemiuiler.utils.Helpers;

import java.util.List;
import java.util.Map;
import java.util.Objects;

import io.luckypray.dexkit.descriptor.member.DexClassDescriptor;

public class GetNumberEnable extends BaseHook {

    Class<?> mLab;
    Class<?> mStableVer;
    Class<?> labUtils;

    @Override
    public void init() {
        try {
            List<DexClassDescriptor> result = Objects.requireNonNull(SecurityCenterDexKit.mSecurityCenterResultClassMap.get("LabUtils"));
            for (DexClassDescriptor descriptor : result) {
                labUtils = descriptor.getClassInstance(lpparam.classLoader);
                log("labUtils class is " + labUtils);
                findAndHookMethod("com.miui.permcenter.settings.PrivacyLabActivity", "onCreateFragment", new MethodHook() {
                    @Override
                    protected void before(MethodHookParam param) throws Throwable {
                        Object fm = Helpers.getStaticObjectFieldSilently(labUtils, "b");
                        if (fm != null) {
                            try {
                                Map<String, Integer> featMap = (Map<String, Integer>) fm;
                                featMap.put("mi_lab_operator_get_number_enable", 0);
                                // featMap.put("mi_lab_blur_location_enable", 0);
                            } catch (Throwable ignore) {
                            }
                        }
                    }
                });
            }
        } catch (Throwable e) {
            e.printStackTrace();
        }

        /*findAndHookMethod(mLab, "f", new MethodHook() {
            @Override
            protected void before(MethodHookParam param) throws Throwable {
                param.setResult(false);
            }
        });*/
        // findAndHookMethod(mStableVer, "IS_STABLE_VERSION", new MethodHook() {
        //    @Override
        //    protected void before(MethodHookParam param) throws Throwable {
        //        param.setResult(true);
        //    }
        //});
    }

}

File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/securitycenter/lab/GetNumberEnable.kt
package com.sevtinge.cemiuiler.module.hook.securitycenter.lab

import com.github.kyuubiran.ezxhelper.EzXHelper
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.module.hook.securitycenter.lab.LabUtilsClass.labUtilClass
import com.sevtinge.cemiuiler.utils.Helpers

object GetNumberEnable : BaseHook() {
    private var labUtils: Class<*>? = null
    override fun init() {
        labUtilClass.forEach {
            labUtils = it.getInstance(EzXHelper.classLoader)
            logI("labUtils class is $labUtils")
            findAndHookMethod(
                "com.miui.permcenter.settings.PrivacyLabActivity",
                "onCreateFragment",
                object : MethodHook() {
                    @Throws(Throwable::class)
                    override fun before(param: MethodHookParam) {
                        val fm = Helpers.getStaticObjectFieldSilently(labUtils, "b")
                        if (fm != null) {
                            try {
                                val featMap = fm as MutableMap<String, Int>
                                featMap["mi_lab_operator_get_number_enable"] = 0
                                // featMap.put("mi_lab_blur_location_enable", 0);
                            } catch (ignore: Throwable) {
                            }
                        }
                    }
                })
        }

        /*   val result: List<DexClassDescriptor> = Objects.requireNonNull<List<DexClassDescriptor>>(
               SecurityCenterDexKit.mSecurityCenterResultClassMap.get("LabUtils")
           )
           for (descriptor in result) {
               labUtils = descriptor.getClassInstance(lpparam.classLoader)
               log("labUtils class is $labUtils")
               findAndHookMethod(
                   "com.miui.permcenter.settings.PrivacyLabActivity",
                   "onCreateFragment",
                   object : BaseHook.MethodHook() {
                       @Throws(Throwable::class)
                       protected override fun before(param: MethodHookParam) {
                           val fm = Helpers.getStaticObjectFieldSilently(labUtils, "b")
                           if (fm != null) {
                               try {
                                   val featMap = fm as MutableMap<String, Int>
                                   featMap["mi_lab_operator_get_number_enable"] = 0
                                   // featMap.put("mi_lab_blur_location_enable", 0);
                               } catch (ignore: Throwable) {
                               }
                           }
                       }
                   })
           }
       } catch (e: Throwable) {
           e.printStackTrace()
       }*/
    }
}
